phpstan.neon

```diff
@@ -0,0 +1,2 @@
+includes:
+    - vendor/phpstan/phpstan-phpunit/extension.neon
```

Comment:
What does this do?

Reply:
It helps with using PHPUnit mocks - PHPStan points out type errors when using mocks and this extension allows mocks when they're specified in the docstring.

Reply:
I've now added PHPUnit-specific rules that'll poke us to use the most expressive assert possible (e.g. `assertCount` and `assertTrue` insetad of using `assertSame` with `count` and boolean values)